### PR TITLE
HealtComponent and death event done

### PR DESCRIPTION
**Pull request by OtakuFreak13**
HC has some temporary function that will print when something reaches 0hp. If you add an enemy with enemyFactory then they will spawn with health. The player in demo scene (Game) has health, and so does the barbs as well. When they attack the player you lose health and eventually print that you are dead.

### CODE REVIEW

File: StortSpelProjekt/Game/src/Gamefiles/Components/HealthComponent.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#ifndef HEALTH_COMPONENT_H
+#define HEALTH_COMPONENT_H
+
+#include "..\..\Engine\src\ECS\Components\Component.h"
+#include "..\Events\Events.h"
```
Comment (kattalainen) [inline StortSpelProjekt/Game/src/Gamefiles/Components/HealthComponent.h]: change \ to / for consistency with rest of the project

Comment (OtakuFreak13) [reply StortSpelProjekt/Game/src/Gamefiles/Components/HealthComponent.h]: done

**Review by kattalainen — DISMISSED** [review]

File: StortSpelProjekt/Game/src/Gamefiles/Components/HealthComponent.h [hunk 2]
```diff
@@ -0,0 +1,44 @@
+#ifndef HEALTH_COMPONENT_H
+#define HEALTH_COMPONENT_H
+
+#include "../../Engine/src/ECS/Components/Component.h"
+#include "../Events/Events.h"
```
Comment (kattalainen) [inline StortSpelProjekt/Game/src/Gamefiles/Components/HealthComponent.h]: change include to forward declaration of eventname, as instructed in the tutorial for events

**Review by kattalainen — DISMISSED** [review]

File: StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp [hunk 3]
```diff
@@ -172,17 +184,18 @@ Entity* EnemyFactory::AddExistingEnemyWithChanges(std::string entityName, float3
 			component::BoundingBoxComponent* bbc = nullptr;
 			component::AccelerationComponent* ac = nullptr;
 			component::AiComponent* ai = nullptr;
-			// TODO: Add more components as they are made such as HealthComponent
 
 			mc = ent->AddComponent<component::ModelComponent>();
 			tc = ent->AddComponent<component::TransformComponent>();
+			ent->AddComponent<component::HealthComponent>(newHP);
```
Comment (fredrix96) [inline StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp:190]: newHp may be unitialized

**Review by fredrix96 — CHANGES_REQUESTED** [review]
Great work! Just change the init of newHP as mentioned

**Review by kattalainen — APPROVED** [review]

**Review by fredrix96 — APPROVED** [review]
Well done 👍

**Pull request merged**